### ISSUE TITLE
Useless transmute lint throws false positive when transmuting lifetimes

### ISSUE THREAD

**Issue by LiamTheProgrammer** [rust-lang/rust-clippy#6899]
This code triggers a false positive for the "useless transmute" lint:

```rust
struct Example<'a> {
    phdata: core::marker::PhantomData<&'a ()>
}

fn useless_transmute_false_positive<'a, 'b>(ex: Example<'a>) -> Example<'b> where 'a: 'b {
    unsafe {
        core::mem::transmute(ex)
    }
}
```

This should not be considered useless because it changes information about the type, but for some reason the type is considered the same by clippy as shown in this message: `` transmute from a type (Example) to itself ``

**Comment by rail-rain** [comment]
This is a duplicate of #5668, which was marked as a duplicate of #5343.

**Comment by y21** [comment]
Issue seems to be fixed, the provided code no longer gets linted: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=81245d0ed74a6d6d671f551285b6d726

**Comment by rail-rain** [comment]
It looks like #8564 fixed this to some degree. It’s probably the reason why the OP’s repro doesn’t work anymore.

Note that the author of #8564 chose not to close #5343 as the fix wasn’t proper. This issue still should be closed though as it’s a duplicate of #5343.
